deploy/deploy_conda.sh

```diff
@@ -1,23 +1,35 @@
 #!/usr/bin/env bash
 
+# quit script if any of the commands fails. Note that && commands should be in paranthesis for this to work.
+set -eo pipefail
+trap 'exit_status="$?" && echo Failed on line: $LINENO at command: $BASH_COMMAND && echo "exit status $exit_status" && exit' ERR
+
+# name for the conda environment. Should be possible to create another name instead of jasen.
 NAME=$1
+# script directory
 scriptdir="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
 
+# If the user did not gave a name for the conda environment on the command line, then the name will be jasen
 if ! [ $NAME ]; then
    NAME=jasen
-fi   
+fi
 
+conda info
+#Unload environment. Deactivation not working properly.
+conda info | grep -q $NAME && conda deactivate || :
+conda info
 
-#Unload environment
-conda info | grep -q $NAME && source deactivate || :
 #Remove environment if already present
 echo "Purging any duplicate existing environment"
 conda remove -y -n $NAME --all || :
 
 echo "Creating JASEN environment named $NAME"
 #conda create --name $NAME -f $scriptdir/reqs/reqs.txt -q 
-conda env create -f deploy/reqs/env-index-mini.yaml 
+# mamba is used for quicker env resolve time
+mamba env create -f deploy/reqs/env-index-mini.yaml 
```

Comment:
This assumes that mamba is already installed, I think it would be better to install mamba in the jasen environemnt first or include it in the `env-index-mini.yaml`...

Reply:
I agree. Here is possible solution.
# first install mamba in a env called jasen
conda create -n jasen -c bioconda mamba
# activate the new env
conda activate jasen
# install the packages defined in the env-index-mini.yaml with mamba in the jasen env.
mamba env update  --file JASEN/deploy/reqs/env-index-mini.yaml